### PR TITLE
Pipelines

### PR DESCRIPTION
**Pull request by trisyoungs**
Add single pipeline for develop branch build / deploy.

### CODE REVIEW

**Comment by DanNixon** [comment]
This isn't quite what I had in mind.
There is a lot of duplication between pipelines and no clear separation between different steps for each platform.

If GH Actions doesn't let you reuse pipeline fragments then I think we'd be better off moving to Azure Pipelines.

**Comment by trisyoungs** [comment]
Only a first step! The duplication between workflows, and the incoherence between OS-specific steps, can largely be removed by splitting into local actions, each dealing with a specific OS.

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        path: installers/
+    - name: Upload Artifacts (OSX)
+      if: matrix.os == 'macos-latest'
+      uses: actions/upload-artifact@v1
+      with:
+        name: artifacts-osx
+        path: damagefiles/
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
```
Comment (DanNixon) [inline .github/workflows/develop.yml]: You likely don't need this, if a previous job fails the entire pipeline does.

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, windows-latest, macos-latest]
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build (Linux)
+      if: matrix.os == 'ubuntu-16.04'
```
Comment (DanNixon) [inline .github/workflows/develop.yml]: Things like this do not help readability at all. See what we do in Scipp.

File: .github/workflows/osx-serial-gui.yml [hunk 3]
```diff
@@ -3,10 +3,10 @@ name: OSX (GUI/Serial)
 on:
   push:
     branches:
-    - develop
-    - master
-    paths:
-    - 'src/**'
+    - NOdevelop
```
Comment (DanNixon) [inline .github/workflows/osx-serial-gui.yml]: If this file is not used simply remove it.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,166 @@
+name: Release Deployment
```
Comment (DanNixon) [inline .github/workflows/release.yml]: How is this different from `develop.yml`?

File: ci/osx/build_serial_gui [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Script to build Dissolve GUI (and serial) versions on OSX systems.
+# Intended to be called as a job within GitHub Actions.
+
+#
+# Configuration
+#
+# -- Enforce verbosity for commands and exit on first failure
+set -ev
```
Comment (DanNixon) [inline ci/osx/build_serial_gui]: Also add `x` here to make build logs a bit more useful.

File: ci/ubuntu/build_16.04_serial_gui [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
```
Comment (DanNixon) [inline ci/ubuntu/build_16.04_serial_gui]: What is specific about Ubuntu 16.04 here? If nothing can the file be renamed.

**Pull request closed without merging**